graphql_compiler/tests/test_compiler.py

```diff
@@ -4835,7 +4966,22 @@ def test_no_op_coercion_inside_fold(self):
                 )
             ])}
         '''
-        expected_sql = NotImplementedError
+        # Not working b/c join descriptors for hyper edges not working properly
+        expected_sql = '''
+        SELECT
+            [Animal_1].name AS animal_name,
+            coalesce(folded_subquery_1.fold_output_1, ARRAY[]::VARCHAR[]) AS related_entities
+        FROM db_1.schema_1.[Animal] AS [Animal_1]
+        LEFT OUTER JOIN (
+            SELECT
+                array_agg([Entity_1].name) AS fold_output_1,
+                [Animal_2].uuid AS uuid
+            FROM db_1.schema_1.[Animal] AS [Animal_2] 
+            JOIN db_1.schema_1.[Entity] AS [Entity_1]
+            ON [Animal_2].uuid = [Entity_1].related_entity
+            GROUP BY [Animal_2].uuid
+        ) AS folded_subquery_1
+        ON [Animal_1].uuid = folded_subquery_1.uuid'''
```

Comment:
```
{
        Animal {
            name @output(out_name: "animal_name")
            out_Entity_Related @fold {
                ... on Entity {
                    name @output(out_name: "related_entities")
                }
            }
        }
    }
```

This fails because hyper edges aren't yet implemented completely so the join descriptor returns the edges in the wrong from/to order for the  Animal -> Entity join